renderdoc/driver/gl/wrappers/gl_draw_funcs.cpp

```diff
@@ -709,10 +709,47 @@ bool WrappedOpenGL::Serialise_glDrawArrays(GLenum mode, GLint first, GLsizei cou
   return true;
 }
 
+void WrappedOpenGL::CopyClientMemoryArrays(GLint first, GLsizei count)
+{
+  GLuint prevBinding;
+  m_Real.glGetIntegerv(eGL_ARRAY_BUFFER_BINDING, (GLint *)&prevBinding);
+
+  for(GLuint i = 0; i < ARRAY_COUNT(m_ClientMemoryVBOs); i++)
+  {
+    GLint enabled, size, stride;
+
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_ENABLED, &enabled);
+    if(!enabled)
+      continue;
+
+    // Check that the active buffer is one of the client-memory copy buffers.
+    GLuint buffer;
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_BUFFER_BINDING, (GLint *)&buffer);
+    if(buffer < m_ClientMemoryVBOs[0] ||
+       buffer > m_ClientMemoryVBOs[ARRAY_COUNT(m_ClientMemoryVBOs) - 1])
+      continue;
+
+    // App initially used client memory, so copy it into the temporary buffer.
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_SIZE, &size);
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_STRIDE, &stride);
+    GLenum type;
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_TYPE, (GLint *)&type);
+
+    GLint totalStride = stride ? stride : (GLint)GLTypeSize(type) * size;
+    glBindBuffer(eGL_ARRAY_BUFFER, buffer);
+    glBufferData(eGL_ARRAY_BUFFER, (first + count) * totalStride, m_ClientMemoryPointers[i],
+                 eGL_STATIC_DRAW);
+  }
+
+  glBindBuffer(eGL_ARRAY_BUFFER, prevBinding);
+}
+
 void WrappedOpenGL::glDrawArrays(GLenum mode, GLint first, GLsizei count)
 {
   CoherentMapImplicitBarrier();
 
+  CopyClientMemoryArrays(first, count);
```

Comment:
Yes I was planning to support glDrawElements after this. I will probably do the same thing and create a temporary buffer for copying indices into the GL_ELEMENT_ARRAY_BUFFER binding.

We also have glDrawElementsIndirect, which will need the GL_DRAW_INDIRECT_BUFFER binding.